<?xml version="1.0"?>
<launch>
  <include file="$(find racecar_simulator)/launch/simulator.launch">
        <arg name="map" value="$(find racecar_simulator)/maps/levine.yaml"/>
  </include>

  <rosparam command="load" file="$(find nautilus_wall_following)/config/params_pure.yaml"/>

  <!-- output="screen" allows us to see print statements-->
  <node pkg="nautilus_scan_matching" name="scan_matcher" type="scan_matching" output="screen"/>
  <node pkg="nautilus_scan_matching" name="scan_quality" type="scan_match_quality" output="screen"/>

  <node pkg="nautilus_wall_following" name="pid_error_node" type="nautilus_pid_error.py"/>	
  <node pkg="nautilus_wall_following" name="control_node" type="nautilus_control.py"/>
  <node pkg="nautilus_wall_following" name="error_analysis" type="nautilus_analysis.py"/>

  <node pkg="wall_following" name="sim_connector_node" type="sim_connector.py" output="screen"/>
  <node pkg="gap_finding" name="visualize_gap_finding_node" type="visualize_gap_finding.py" output="screen"/>

</launch>
